src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState } from 'react';
+import { useMatch } from 'react-router-dom';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+
+  const match = useMatch('/people/:personSlug');
+  const selectedPersonSlug = match?.params.personSlug;
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then((result) => {
+        setIsLoading(false);
+        setPeople(result.map(person => ({
+          ...person,
+          mother: result
+            .find(mother => mother.name === person.motherName),
+          father: result
+            .find(father => father.name === person.fatherName),
+        })));
+      })
+      .catch(() => {
+        setHasError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const noDataError = !isLoading && people.length === 0;
```

Comment:
Great job implementing a varialbe to save the conditions.

Reply:
yeah, agree with @Grayfox147 
But comparison with 0 looks kinda hardcoded, we usually do like that
```suggestion
  const noDataError = !isLoading && !people.length;
```